src/core/postNote.ts

```diff
@@ -50,47 +36,36 @@ export class PostNote {
         });
     }
 
-    async post(emailContent: EmailContent, folders: string[], tags: string []): Promise<void> {
-        if (folders.length === 0) return;
-
+    async post(emailContent: EmailContent, folderId: string, tags: string []): Promise<void> {
         this.attachments = emailContent.attachments;
         this.subject = emailContent.subject;
 
         // to get only the HTML body of the email and ignore unnecessary data.
         this.emailHtmlBody = new DOMParser().parseFromString(emailContent.html, 'text/html').body.innerHTML;
 
         // Converting email to (html, markdown, text) depends on what export type is in the plugin settings.
-        await this.converter();
+        await this.convert();
 
-        await this.addFolders(folders);
         await this.addTags(tags);
 
-        const joplinFolders = await joplin.data.get(['folders']);
         const joplinTags = await joplin.data.get(['tags']);
 
-        const emailFolders = joplinFolders.items.filter((e)=>{
-            return this.folders.includes(e.title);
-        });
-
         const emailTags = joplinTags.items.filter((e)=>{
             return this.tags.includes(e.title);
```

Comment:
I can't find how these tags are set right now, but I think this will stop working if a tag is renamed. Best to use IDs as you did with folder.

I guess it's not a concern for manual one-time import but for monitoring mailbox it is.

Reply:
I moved this code to the 'addTags' function and fixed it so that if 'has_more' in Joplin tags is true, this function now takes a set of tags and creates all tags that are not inside Joplin tags, and it returns all mentioned tags with all properties. 

If something is not clear, please tell me.